// @ston-fi/sdk/dist/chunk-MKM65SV2.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;

var _chunk6OWOA27Rcjs = require('./chunk-6OWOA27R.cjs');


var _chunkNSMGSBVJcjs = require('./chunk-NSMGSBVJ.cjs');


var _chunk6ZI7IZKLcjs = require('./chunk-6ZI7IZKL.cjs');


var _chunkWF6EZ2CWcjs = require('./chunk-WF6EZ2CW.cjs');


var _chunk666S76J6cjs = require('./chunk-666S76J6.cjs');

// src/contracts/dex/v2_2/router/CPIRouterV2_2.ts
var CPIRouterV2_2 = (_class = class extends _chunk6ZI7IZKLcjs.CPIRouterV2_1 {
  static __initStatic() {this.version = _chunk666S76J6cjs.DEX_VERSION.v2_2}
  async getPool(provider, params) {
    const poolAddress = await this.getPoolAddressByJettonMinters(
      provider,
      params
    );
    return _chunkNSMGSBVJcjs.CPIPoolV2_2.create(poolAddress);
  }
  async getVault(provider, params) {
    const tokenMinter = provider.open(_chunkWF6EZ2CWcjs.JettonMinter.create(params.tokenMinter));
    const vaultAddress = await this.getVaultAddress(provider, {
      user: params.user,
      tokenWallet: await tokenMinter.getWalletAddress(this.address)
    });
    return _chunk6OWOA27Rcjs.VaultV2_2.create(vaultAddress);
  }
}, _class.__initStatic(), _class);



exports.CPIRouterV2_2 = CPIRouterV2_2;
