// @ston-fi/sdk/dist/chunk-UZRN4ZRE.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;

var _chunk6OWOA27Rcjs = require('./chunk-6OWOA27R.cjs');


var _chunkDUZEEI45cjs = require('./chunk-DUZEEI45.cjs');


var _chunkKJI252NXcjs = require('./chunk-KJI252NX.cjs');


var _chunkWF6EZ2CWcjs = require('./chunk-WF6EZ2CW.cjs');


var _chunk666S76J6cjs = require('./chunk-666S76J6.cjs');

// src/contracts/dex/v2_2/router/BaseRouterV2_2.ts
var BaseRouterV2_2 = (_class = class extends _chunkKJI252NXcjs.BaseRouterV2_1 {
  static __initStatic() {this.version = _chunk666S76J6cjs.DEX_VERSION.v2_2}
  async getPool(provider, params) {
    const poolAddress = await this.getPoolAddressByJettonMinters(
      provider,
      params
    );
    return _chunkDUZEEI45cjs.BasePoolV2_2.create(poolAddress);
  }
  async getVault(provider, params) {
    const tokenMinter = provider.open(_chunkWF6EZ2CWcjs.JettonMinter.create(params.tokenMinter));
    const vaultAddress = await this.getVaultAddress(provider, {
      user: params.user,
      tokenWallet: await tokenMinter.getWalletAddress(this.address)
    });
    return _chunk6OWOA27Rcjs.VaultV2_2.create(vaultAddress);
  }
}, _class.__initStatic(), _class);



exports.BaseRouterV2_2 = BaseRouterV2_2;
